pennylane_forest/qpu.py

```diff
@@ -66,28 +61,21 @@ class QPUDevice(QVMDevice):
         readout_error (list): specifies the conditional probabilities [p(0|0), p(1|1)], where
             p(i|j) denotes the prob of reading out i having sampled j; can be set to `None` if no
             readout errors need to be simulated; can only be set for the QPU-as-a-QVM
-        symmetrize_readout (str): method to perform readout symmetrization, using exhaustive
+        symmetrize_readout (pyquil.experiment.SymmetrizationLevel): method to perform readout symmetrization, using exhaustive
             symmetrization by default
         calibrate_readout (str): method to perform calibration for readout error mitigation,
             normalizing by the expectation value in the +1-eigenstate of the observable by default
 
     Keyword args:
-        forest_url (str): the Forest URL server. Can also be set by
-            the environment variable ``FOREST_SERVER_URL``, or in the ``~/.qcs_config``
-            configuration file. Default value is ``"https://forest-server.qcs.rigetti.com"``.
-        qvm_url (str): the QVM server URL. Can also be set by the environment
-            variable ``QVM_URL``, or in the ``~/.forest_config`` configuration file.
-            Default value is ``"http://127.0.0.1:5000"``.
-        compiler_url (str): the compiler server URL. Can also be set by the environment
-            variable ``COMPILER_URL``, or in the ``~/.forest_config`` configuration file.
-            Default value is ``"http://127.0.0.1:6000"``.
-        timeout (int): number of seconds to wait for a response from the client.
+        compiler_timeout (int): number of seconds to wait for a response from quilc (default 10).
+        execution_timeout (int): number of seconds to wait for a response from the QVM (default 10).
         parametric_compilation (bool): a boolean value of whether or not to use parametric
             compilation.
     """
     name = "Forest QPU Device"
     short_name = "forest.qpu"
     observables = {"PauliX", "PauliY", "PauliZ", "Identity", "Hadamard", "Hermitian"}
+    pennylane_requires = ">=0.15.0"
 
     def __init__(
```

Comment:
The `QPUDevice.__init__` contains duplicated code from the `QVMDevice.__init__`

Reply:
When calling super, use:
```python
super().__init__(wires, shots, **kwargs)
```

Instead of:
```python
super(QVMDevice, self).__init__(wires, shots, **kwargs)
```

Reply:
The way `super` is being used in `QPUDevice` is related to why the code is duplicated between `QPUDevice.__init__` and `QVMDevice.__init__`. 

For example, `QPUDevice` and `QVMDevice` initialize `self.qc` differently. If `QPUDevice` were to call `QVMDevice.__init__`, then it would effectively clobber the `qc` it had setup for itself. Because of this, it bypasses the `QVMDevice` by calling it's superclasses initializer (`ForestDevice.__init__`):

```python
super(QVMDevice, self).__init__(wires, shots, **kwargs)
```

Since they can't share an initializer, `QPUDevice` also has to duplicate all the stuff about initialization it shares in common with `QVMDevice`

As discussed in another thread, this is why `QPUDevice` shouldn't inherit from `QVMDevice`. The right move here is to refactor the shared logic into `ForestDevice` that both devices inherit from. I've gone ahead and started down that path since it's likely to cause more confusion until it's cleaned up.

Reply:
Thanks for the explanation! I didn't realise we were calling `super(QVMDevice)` instead of `super(QPUDevice)`.

Reply:
Then we can also remove the `kwargs` from here 😄